Dependencies/xr/Source/ARCore/XR.cpp

```diff
@@ -419,6 +436,84 @@ namespace xr
         ArCamera_release(camera);
     }
 
+    void System::Session::Frame::GetHitTestResults(std::list<Pose>& filteredResults, xr::Ray offsetRay) const {
+        ArSession* session = m_impl->sessionImpl.session;
+        ArCamera* camera{};
+        ArFrame_acquireCamera(session, m_impl->sessionImpl.frame, &camera);
+
+        // Get the tracking state
+        ArTrackingState trackingState{};
+        ArCamera_getTrackingState(session, camera, &trackingState);
+
+        // If not tracking, back out and return.
+        if (trackingState != ArTrackingState::AR_TRACKING_STATE_TRACKING) {
+            return;
+        }
+
+        // Push the camera orientation into a glm quaternion.
+        glm::quat cameraOrientationQuaternion;
+        cameraOrientationQuaternion.x = Views[0].Space.Pose.Orientation.X;
+        cameraOrientationQuaternion.y = Views[0].Space.Pose.Orientation.Y;
+        cameraOrientationQuaternion.z = Views[0].Space.Pose.Orientation.Z;
+        cameraOrientationQuaternion.w = Views[0].Space.Pose.Orientation.W;
+
+        // Pull out the direction from the offset ray into a GLM Vector3.
+        glm::vec3 direction = { offsetRay.Direction.X, offsetRay.Direction.Y, offsetRay.Direction.Z};
+
+        // Multiply the camera rotation quaternion by the direction vector to calculate the direction vector in viewer space.
+        glm::vec3 cameraOrientedDirection = cameraOrientationQuaternion * glm::normalize(direction);
+        float cameraOrientedDirectionArray[3] = { cameraOrientedDirection.x, cameraOrientedDirection.y, cameraOrientedDirection.z };
+
+        // Convert the origin to camera space by multiplying the origin by the rotation quaternion, then adding that to the
+        // position of the camera.
+        glm::vec3 offsetOrigin = { offsetRay.Origin.X, offsetRay.Origin.Y, offsetRay.Origin.Z};
+        offsetOrigin = cameraOrientationQuaternion * offsetOrigin;
+
+        // Pull out the origin composited from the offsetRay and camera position into a float array.
+        float hitTestOrigin[3] = {
+                Views[0].Space.Pose.Position.X + offsetOrigin.x,
+                Views[0].Space.Pose.Position.Y + offsetOrigin.y,
+                Views[0].Space.Pose.Position.Z + offsetOrigin.z
+        };
+
+        // Perform a hit test and process the results.
+        ArFrame_hitTestRay(session, m_impl->sessionImpl.frame, hitTestOrigin, cameraOrientedDirectionArray, m_impl->sessionImpl.hitResultList);
+
+        // Iterate over the results and pull out only those that match the desired TrackableType (just Planes for now)
```

Comment:
Yes see: https://immersive-web.github.io/hit-test/#hit-test-trackable-type-enum

This is the equivalent of saying XRHitTestTrackableType.Mesh since the current requirement is in the detected plane polygon.  I left them defaulted to meshes right now, since we don't actually expose the hook to set this in Babylon.js yet.